src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4713 @@
+    }
+    else if (bVar20) {
+        m0366 = 0;
+        sVar26 = 0;
+    }
+    
+    if (!checkStateFlg(daSFLG_UNK1_e)) {
+        if (!(m034C == 12 || m034C == 15 || m034C == 7 || m034C == 11 || m034C == 13 || m034C == 16 || dComIfGp_event_runCheck())) {
+            if (mTornadoActor) {
+                sVar16 = shape_angle.y;
+                cLib_addCalcAngleS(&shape_angle.y, m040C * 10430.378f + 20480.0f, 5, 0x2000, 0x200);
+                setControllAngle(getAimControllAngle(sVar16));
+                current.angle.y = shape_angle.y;
+                
+            }
+            else {
+                if (mWhirlActor) {
+                    sVar16 = shape_angle.y;
+                    if (m0352) {
+                        sVar5 = m040C * 10430.378f + 20480.0f;
+                    }
+                    else {         
+                        sVar5 = m040C * 10430.378f + 32768.0f;
+                    }
+                    cLib_addCalcAngleS(&shape_angle.y, sVar5, 5, 0x2000, 0x200);
+                    setControllAngle(getAimControllAngle(sVar16));
+                    current.angle.y = shape_angle.y;
+                }
+            }
+        }
+        sVar16 = -m0384;
+        sVar5 = sVar16 * 0.05f;
+        if (sVar5 == 0) {
+            if (sVar16 > 0) {
+                sVar5 = 1;
+            }
+            else if (sVar16 < 0) {
+                sVar5 = -1;
+            }
+        }
+        m0386 += sVar5;
+        m0384 += m0386;
+        cLib_addCalcAngleS(&m0386, 0, 30, 0x1000, 1);
+    }
+    else {
+        cLib_addCalcAngleS(&m0384, -0x800, 3, 0x500, 0x300);
+    }
+
+    m045C = current.pos;
+
+    if (checkStateFlg(daSFLG_UNK2000000_e) == 0) {
+        cLib_chaseS(&m03B8, 0, 0x80);
+    }
+    else {
+        cLib_chaseS(&m03B8, 0x4E8, 0x80);
+    }
+
+    float fVar4;
+    if (!(m034C == 12 || m034C == 15)) {
+        if (!(m034C == 7 || m034C == 11)) {
+            if (!(!dComIfGp_event_runCheck() && !daPy_getPlayerLinkActorClass()->checkNoControll() || mEvtStaffId != -1 || m034C == 8 || m034C == 16 || checkStateFlg(daSFLG_UNK100000_e))) {
+                if (checkStateFlg(daSFLG_UNK20000000_e)) {
+                    if (speedF >= 10.0f) {
+                        firstDecrementShipSpeed(10.0f);
+                    }
+                }
+                else {
+                    speedF = 0.0f;
+                }
+            }
+            if (!dComIfGp_event_runCheck() && !checkStateFlg(daSFLG_UNK1_e) && checkForceMove()) {
+                speedF = fVar3;
+                if (mTornadoActor) {
+                    fVar4 = m0404 * 40.0f + 30.0f;
+                    if (!dComIfGp_checkPlayerStatus0(0,0x1000) || !dComIfGp_checkCameraAttentionStatus(0,0x20)) {
+                        fVar4 *= 1.2f;
+                    }
+                }
+                else {
+                    fVar4 = m0404 * 30.0f + 10.0f;
+                    if (!dComIfGs_getBombNum() && fopAcM_GetRoomNo(this) == 0x2C) {
```

Comment:
And the opposite, this doesn't return a boolean
```suggestion
                    if (dComIfGs_getBombNum() == 0 && fopAcM_GetRoomNo(this) == 0x2C) {
```
Generally if a function has "is" in its name it probably returns a bool, if it has "num" it probably returns a count